splice/web/api/reporting.py

```diff
@@ -1,34 +1,75 @@
 from flask import Blueprint
-from flask_restful import Api, Resource, marshal, fields
-from splice.queries.reporting import get_stats_by_date
+from flask_restful import Api, Resource, marshal, fields, reqparse
+from splice.queries.reporting import get_stats
 
 
 reporting_bp = Blueprint('api.reporting', __name__, url_prefix='/api')
 api = Api(reporting_bp)
 
-reporting_fields = {
+valid_group_by = {
     'date': fields.DateTime(dt_format='iso8601'),
-    'impressions': fields.Integer,
-    'clicks': fields.Integer,
-    'pinned': fields.Integer,
-    'blocked': fields.Integer
+    'week': fields.Integer,
+    'month': fields.Integer,
+    'category': fields.String,
+    'locale': fields.String,
+    'country_code': fields.String
 }
 
+valid_filters = {
+    'type': {'choices': ('affiliate', 'sponsored'), 'store_missing': False},
```

Comment:
A constant "type" has been defined in the `models.Tiles` class, we can use that here. The same to the "adgroup_type" below.